src/Parser/Tokenizer/LookAhead.php

```diff
@@ -58,7 +58,9 @@ public function getIterator(): \Iterator
             yield $token;
         }
 
-        yield from $this->tokens;
+        if (!Scanner::isEnd($this->tokens)) {
+            yield from $this->tokens;
+        }
```

Comment:
```suggestion
        yield from $this->tokens;
```

`Scanner::isEnd` only checks whether `$this->tokens` is still valid. This is implied in `yield from` - the extra check is not needed.